Manage HTML entities in timed text tracks (#552)
openfun/marsha: UPGRADE.md

The format is inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 3.0.0 to 3.0.1

### Before switching

> - AWS lambda-encode must be deploy before switching to this version.

Reviewer: ```suggestion
- AWS lambda-encode must be deployed before switching to this version.
```

> - AWS lambda-encode must be deployed before switching to this version.

Reviewer 2: re-deployed or updated?

Author: For each release we made all lambda must be deployed. Here it is important to deploy them before running the migration.

> - A migration will be run. This migration will look to all existing timed text tracks in the

Reviewer 2: ```suggestion
- A Django data migration will be run that will look to all existing timed text tracks in the
```

>   S3 bucket source and trigger the marsha-encode lambda to escape them. If you don't want to

Reviewer 2: ```suggestion
  S3 source bucket and trigger the marsha-encode lambda to escape them. If you don't want to`

  run this migration you should set `SHOULD_RUN_TIMEDTEXT_TRACKS_ESCAPING_MIGRATION` setting

>   to False. We want to warn that not running this migration is strongly discouraged.

Reviewer 2: ```suggestion
  to False. We want to warn that not running this migration is strongly discouraged and poses security threats as the content of the timed text tracks will now be displayed insecurely by React.
```

The format is inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 3.0.0 to 3.1.0

### Before deploying

- new setting `AWS_BASE_NAME`. The value should be the terraform workspace name used to deploy
  all the AWS stack. For example if your workspace name is `production` set `AWS_BASE_NAME` to
  `production`

> - AWS lambda-encode must be deployed before running django migration.

Reviewer 2: ```suggestion
- AWS lambda-encode must be deployed before running Django migrations.
```
